pkg/render/apiserver.go

```diff
@@ -279,6 +279,9 @@ func (c *apiServerComponent) Objects() ([]client.Object, []client.Object) {
 		// Explicitly delete any global OSS objects.
 		// Namespaced objects will be handled by namespace deletion.
 		objsToDelete = append(objsToDelete, globalCalicoObjects...)
+
+		// Explicitly delete any renamed/deprecated objects.
+		objsToDelete = append(objsToDelete, c.getDeprecatedResources()...)
```

Comment:
I think this can be outside (under) the if-else statement. This always needs to get deleted, regardless of whether the cluster is enterprise right now.

Reply:
I moved it outside, but for opensource we dont have the "tigera-crds" clusterrole and the apiserver binding. Doesn't it matter or wouldn't it fail if they dont exist?